File: BUNKCALC.c
#include <stdio.h>
int main() {
    char subject[50];
    int tc; //tc matlab total class

    printf("\n\t\t\t\tBUNK CALCULATOR\n\n");
    printf("\t\t\4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4\n");
    printf("\t\t\4\t\t\t\t\t\t\4\n");
    printf("\t\t\4\t\t\t\t\t\t\4\n");
    printf("\t\t\4\tWELCOME TO BUNK CALCULATOR\t\t\4\n");
    printf("\t\t\4\t\t\t\t\t\t\4\n");
    printf("\t\t\4\t\t\t\t\t\t\4\n");
    printf("\t\t\4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4 \4\n");

    // Input  from user subject name
    printf("Enter the subject name: ");
    scanf("%s", subject);

    // Input total number of classes(Subject teacher se pucho)
    printf("Enter the total number of classes: ");
    scanf("%d", &tc);

    // Calculate the maximum number of classes that can be missed and bunked for each percentage
    int mcm_95 = tc - (tc * 0.95);
    int mcm_90 = tc - (tc * 0.90);
    int mcm_85 = tc - (tc * 0.85);
    int mcm_80 = tc - (tc * 0.80);
    int mcm_75 = tc - (tc * 0.75);

    // output
    printf("Subject: %s\n", subject);
    printf("Total Classes: %d\n", tc);
    printf("You can miss/ bunk a maximum of:\n");
    printf("- %d classes for attendance above 95%%\n", mcm_95);
    printf("- %d classes for attendance above 90%%\n", mcm_90);
    printf("- %d classes for attendance above 85%%\n", mcm_85);
    printf("- %d classes for attendance above 80%%\n", mcm_80);
    printf("- %d classes for attendance above 75%%\n", mcm_75);

    printf("\n\t\t\t\t    Thank you! \n");
    printf("\t\t\t   FOR USING BUNK CALC\n");
    printf("\t\t\t    www.BUNKMATKAROFAILHOJAOGE.com\n");


    return 0;
}

File: bunkcalc.py
import tkinter as tk
from tkinter import messagebox

def calculate_bunks():
    subject = subject_entry.get()
    try:
        total_classes = int(total_classes_entry.get())
        
        # Calculate the maximum number of classes that can be missed
        mcm_95 = int(total_classes - (total_classes * 0.95))
        mcm_90 = int(total_classes - (total_classes * 0.90))
        mcm_85 = int(total_classes - (total_classes * 0.85))
        mcm_80 = int(total_classes - (total_classes * 0.80))
        mcm_75 = int(total_classes - (total_classes * 0.75))

        # Display the results
        result_text = (
            f"Subject: {subject}\n"
            f"Total Classes: {total_classes}\n"
            "You can miss/ bunk a maximum of:\n"
            f"- {mcm_95} classes for attendance above 95%\n"
            f"- {mcm_90} classes for attendance above 90%\n"
            f"- {mcm_85} classes for attendance above 85%\n"
            f"- {mcm_80} classes for attendance above 80%\n"
            f"- {mcm_75} classes for attendance above 75%"
        )
        
        messagebox.showinfo("Bunk Calculator Result", result_text)
    except ValueError:
        messagebox.showerror("Input Error", "Please enter a valid number for total classes.")

# Create the main window
root = tk.Tk()
root.title("Bunk Calculator")

# Create and place the widgets
tk.Label(root, text="BUNK CALCULATOR", font=("Arial", 16)).pack(pady=10)

tk.Label(root, text="Enter the subject name:").pack(pady=5)
subject_entry = tk.Entry(root)
subject_entry.pack(pady=5)

tk.Label(root, text="Enter the total number of classes:").pack(pady=5)
total_classes_entry = tk.Entry(root)
total_classes_entry.pack(pady=5)

calculate_button = tk.Button(root, text="Calculate Bunks", command=calculate_bunks)
calculate_button.pack(pady=20)

# Run the application
root.mainloop()
